src/coreclr/tools/Common/JitInterface/CorInfoInstructionSet.cs

```diff
@@ -294,6 +294,35 @@ public void ExpandInstructionSetByImplication(TargetArchitecture architecture)
             this = ExpandInstructionSetByImplicationHelper(architecture, this);
         }
 
+        public static InstructionSet ConvertToImpliedInstructionSetForVectorInstructionSets(TargetArchitecture architecture, InstructionSet input)
+        {
+            switch(architecture)
+            {
+            case TargetArchitecture.ARM64:
+                switch(input)
+                {
+                case InstructionSet.ARM64_Vector64: return InstructionSet.ARM64_AdvSimd;
+                case InstructionSet.ARM64_Vector128: return InstructionSet.ARM64_AdvSimd;
+                }
+                break;
+            case TargetArchitecture.X64:
+                switch(input)
+                {
+                case InstructionSet.X64_Vector128: return InstructionSet.X64_SSE;
+                case InstructionSet.X64_Vector256: return InstructionSet.X64_AVX;
+                }
+                break;
+            case TargetArchitecture.X86:
+                switch(input)
+                {
+                case InstructionSet.X86_Vector128: return InstructionSet.X86_SSE;
+                case InstructionSet.X86_Vector256: return InstructionSet.X86_AVX;
```

Comment:
should `Vector<T>` be normalized on the same handling?

Should there be a comment covering that:
1. The ABI handling for Vector128/256 requires Sse/Avx, this is what's required for them to be passed around as `__m128` and `__m256`, respectively
2. The `Vector128/256.IsHardwareAccelerated` property only returns `true` under Sse2/Avx2 (this is when users can depend on most functions exposed to be SIMD accelerated)

Reply:
Abi handling is actually separate from this entirely and is handled by an abi stability protection construct which works independently of all of this. (Mostly as the ABI for both Vector128 and Vector256 doesn't match the native vector ABI on Windows, and there was a desire at one point to unify on the actual official ABI. This has not yet happened.

Reply:
So is this check only for `IsHardwareAccelerated` then?

Reply:
Just trying to understand what these particular checks line up with.

We support efficiently passing and some acceleration for `Vector128<T>` under `SSE` and `Vector256<T>` under `AVX`.
However, `IsHardwareAccelerated` will only return `true` under `SSE2` and `AVX2`, respectively.

Reply:
This lines up with the Vector128/Vector256 api usage in the JIT. Effectively, when the jit determines that it cannot use a  VectorXXX api it will call up, and notify crossgen that It has determined that if VectorXXX is useable at runtime via any intrinsics the generated code cannot be used. At that stage crossgen needs to encode that one of the instruction sets is not permitted, as it cannot encode that the VectorXXX intrinsics are not available. So, what it does it encode that the minimum required instruction set for the VectorXXX intrinsics is not available. Handling of IsHardwareAccelerated is actually a side-effect of this. In the encoding there are actually more states for how this works than is pleasant to contemplate. For instance, the options available to crossgen for encoding a method which calls `Vector256<T>.IsHardwareAccelerated` are:

1. Generate a method where IsHardwareAccelerated returns false, and mark that if Avx is supported, then the method cannot be used. (In this case the JIT will specify that support for the Vector256 instruction set is required to be disabled)
2. Generate a method where IsHardwareAccelerated returns false, and mark that if Avx2 is supported, then the method cannot be used, and also mark that Avx must be supported to use the method. (In this case the JIT will specify that support for the Vector256 instruction set is required to be enabled, but Avx2 support is required to be disabled.)
3. Generate a method where IsHardwareAccelerated returns true, and mark that if Avx2 is not supported then the method cannot be used. (In this case the JIT will specify that support for the Vector256 instruction set is required to be enabled, and Avx2 support is required to be enabled.)

The problem that this bug solves is that while the previous logic correctly handled cases 2 and 3, it did not handle case 1. (What would happen is that the JIT would notify crossgen that Vector256 could not be used, and be we would fail to record that there was an instruction set restriction on the generated code.)